<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Air Quality Neural Network</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
        <widget class="QLabel" name="headerImageLabel">
            <property name="minimumSize">
                <size>
                    <width>800</width>
                    <height>150</height>
                </size>
            </property>
            <property name="maximumSize">
                <size>
                    <width>16777215</width>
                    <height>150</height>
                </size>
            </property>
            <property name="styleSheet">
                <string notr="true">background-color: #3498db;
                    border-bottom: 2px solid #2980b9;
                    qproperty-alignment: AlignCenter;</string>
            </property>
            <property name="text">
                <string>&lt;h1 style="color: white;"&gt;Air Quality Neural Network&lt;/h1&gt;
                &lt;p style="color: white;"&gt;Predicting air quality using Artificial Neural Network&lt;/p&gt;</string>
            </property>
            <property name="textFormat">
                <enum>Qt::RichText</enum>
            </property>
        </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <widget class="QWidget" name="tabTraining">
       <attribute name="title">
        <string>Training</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Epochs:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="epochSpinBox">
            <property name="minimum">
             <number>100</number>
            </property>
            <property name="maximum">
             <number>10000</number>
            </property>
            <property name="value">
             <number>1000</number>
            </property>
            <property name="singleStep">
             <number>100</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="trainButton">
            <property name="text">
             <string>Train Model</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="plotLabel">
          <property name="frameShape">
           <enum>QFrame::Box</enum>
          </property>
          <property name="text">
           <string>Training plot will appear here</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
            <widget class="QLabel" name="accuracyLabel">
                <property name="text">
                    <string>Training Accuracy: -</string>
                    </property>
                    <property name="alignment">
                    <set>Qt::AlignCenter</set>
                    </property>
                    <property name="styleSheet">
                    <string notr="true">font-size: 14px; 
                                        font-weight: bold; 
                                        color: #27ae60;
                                        margin-top: 10px;</string>
                </property>
            </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabPrediction">
       <attribute name="title">
        <string>Prediction</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>PM10:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QDoubleSpinBox" name="pm10SpinBox">
            <property name="decimals">
             <number>2</number>
            </property>
            <property name="maximum">
             <number>9999.99</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>SO2:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="so2SpinBox">
            <property name="decimals">
             <number>2</number>
            </property>
            <property name="maximum">
             <number>9999.99</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>CO:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QDoubleSpinBox" name="coSpinBox">
            <property name="decimals">
             <number>2</number>
            </property>
            <property name="maximum">
             <number>9999.99</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>O3:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QDoubleSpinBox" name="o3SpinBox">
            <property name="decimals">
             <number>2</number>
            </property>
            <property name="maximum">
             <number>9999.99</number>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>NO2:</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QDoubleSpinBox" name="no2SpinBox">
            <property name="decimals">
             <number>2</number>
            </property>
            <property name="maximum">
             <number>9999.99</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="predictButton">
          <property name="text">
           <string>Predict Air Quality</string>
          </property>
          <property name="styleSheet">
           <string notr="true">font-weight: bold;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="resultLabel">
          <property name="text">
           <string>Predicted Category: -</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="styleSheet">
           <string notr="true">font-size: 18px; font-weight: bold; color: #2c3e50;</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Probabilities:</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="baikProbLabel">
          <property name="text">
           <string>BAIK: 0%</string>
          </property>
          <property name="styleSheet">
           <string notr="true">color: #27ae60;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="sedangProbLabel">
          <property name="text">
           <string>SEDANG: 0%</string>
          </property>
          <property name="styleSheet">
           <string notr="true">color: #f39c12;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="tidakSehatProbLabel">
          <property name="text">
           <string>TIDAK SEHAT: 0%</string>
          </property>
          <property name="styleSheet">
           <string notr="true">color: #e74c3c;</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>